res/braveDefaults.nsi

```diff
@@ -1,137 +1,138 @@
-    WriteRegStr SHCTX "SOFTWARE\Clients\StartMenuInternet\Brave" "" "Brave"
-    WriteRegStr SHCTX "SOFTWARE\Clients\StartMenuInternet\Brave\Capabilities" "ApplicationDescription" "Brave is the new and fast web browser that protects your privacy and security by blocking intrusive ads and trackers."
-    WriteRegStr SHCTX "SOFTWARE\Clients\StartMenuInternet\Brave\Capabilities" "ApplicationName" "Brave"
-    WriteRegDWORD SHCTX "SOFTWARE\Clients\StartMenuInternet\Brave\InstallInfo" "IconsVisible" 1
-    WriteRegStr SHCTX "SOFTWARE\Clients\StartMenuInternet\Brave\shell\open\command" "" ""
-    ; File associations
-    WriteRegStr SHCTX "SOFTWARE\Clients\StartMenuInternet\Brave\Capabilities\FileAssociations" ".htm" "BraveHTML"
-    WriteRegStr SHCTX "SOFTWARE\Clients\StartMenuInternet\Brave\Capabilities\FileAssociations" ".html" "BraveHTML"
-    WriteRegStr SHCTX "SOFTWARE\Clients\StartMenuInternet\Brave\Capabilities\FileAssociations" ".shtml" "BraveHTML"
-    WriteRegStr SHCTX "SOFTWARE\Clients\StartMenuInternet\Brave\Capabilities\FileAssociations" ".xht" "BraveHTML"
-    WriteRegStr SHCTX "SOFTWARE\Clients\StartMenuInternet\Brave\Capabilities\FileAssociations" ".xhtml" "BraveHTML"
-    WriteRegStr SHCTX "SOFTWARE\Clients\StartMenuInternet\Brave\Capabilities\FileAssociations" ".webp" "BraveHTML"
-    ; Protocol associations
-    WriteRegStr SHCTX "SOFTWARE\Clients\StartMenuInternet\Brave\Capabilities\URLAssociations" "ftp" "BraveHTML"
-    WriteRegStr SHCTX "SOFTWARE\Clients\StartMenuInternet\Brave\Capabilities\URLAssociations" "http" "BraveHTML"
-    WriteRegStr SHCTX "SOFTWARE\Clients\StartMenuInternet\Brave\Capabilities\URLAssociations" "https" "BraveHTML"
-    WriteRegStr SHCTX "SOFTWARE\Clients\StartMenuInternet\Brave\Capabilities\URLAssociations" "mailto" "BraveHTML"
-    ; Uninstall icon
-    SetRegView 64
-    WriteRegStr HKCU "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Brave" "DisplayIcon" "$BraveIconPath"
-  ${EndIf}
-SectionEnd
+!include "MUI2.nsh"
+!include "LogicLib.nsh"
+!include "Win8WinVer.nsh"
+!include "GetParameters.nsh"
+!include "GetParent.nsh"
+!include "StrStr.nsh"
+
+!addplugindir "."
+!include "UAC.nsh"
+
+!define MUI_ICON "app.ico"
+!insertmacro MUI_LANGUAGE "English"
+
+SilentInstall silent
+
+Name "Brave"
+OutFile "../Brave-win32-${ARCH}/resources/BraveDefaults.exe"
+RequestExecutionLevel user
+Var BraveEXEPath
+Var BraveIconPath
+Var IsElevated
+Var IsUninstall
+
+Function .onInit
+  ; The StartMenuInternet key can be set in HKCU on Win8 and above.
+  ${IfNot} ${AtLeastWin8}
+    SetShellVarContext all
+  ${EndIf}
+
+  ; Determine if we're elevated currently
+  ClearErrors
+  WriteRegStr HKLM "Software\Brave" "InstallerTest" "Write Test"
+  DeleteRegValue HKLM "Software\Brave" "InstallerTest"
+  ${If} ${Errors}
+    StrCpy $IsElevated "0"
+  ${Else}
+    StrCpy $IsElevated "1"
+  ${EndIf}
+
+  Push $EXEDIR
+  Call GetParent
+  Call GetParent
```

Comment:
Sorry for the confusing diff; This is the fix here (calling `GetParent` twice)

I messed around with my .gitconfig for about 20 minutes trying it to only recognize this one line change before giving up and just checking it in like this. I believe this file in particular was checked in **_with_** CRLF (and I'm not about to change my config to do that). Good news: future updates won't have this conflict